src/tiled/mainwindow.cpp

```diff
@@ -1509,6 +1497,99 @@ void MainWindow::retranslateUi()
+                                                        this);
+    if (!exportDetails.valid()) {
+        return;
+    }
+
+    // Check if writer will overwrite existing files here because some writers
+    // could save to multiple files at the same time. For example CSV saves
+    // each layer into a separate file.
+    QStringList outputFiles = exportDetails.mFormat->outputFiles(mapDocument->map(), exportDetails.mFileName);
+    if (outputFiles.size() > 0) {
+        // Check if any output file already exists
+        QString message =
+                tr("Some export files already exist:") + QLatin1String("\n\n");
+
+        bool overwriteHappens = false;
+
+        for (const QString &outputFile : outputFiles) {
+            if (QFile::exists(outputFile)) {
+                overwriteHappens = true;
+                message += outputFile + QLatin1Char('\n');
+            }
+        }
+        message += QLatin1Char('\n') + tr("Do you want to replace them?");
+
+        // If overwrite happens, warn the user and get confirmation before exporting
+        if (overwriteHappens) {
+            const QMessageBox::StandardButton reply = QMessageBox::warning(
+                                                          this,
+                                                          tr("Overwrite Files"),
+                                                          message,
+                                                          QMessageBox::Yes | QMessageBox::No,
+                                                          QMessageBox::No);
+
+            if (reply != QMessageBox::Yes)
+                return;
+        }
+    }
+
+    Preferences *pref = Preferences::instance();
+
+    pref->setLastPath(Preferences::ExportedFile, QFileInfo(exportDetails.mFileName).path());
+    mSettings.setValue(QLatin1String("lastUsedExportFilter"), selectedFilter);
+
+    auto exportResult = exportDetails.mFormat->writeMap(mapDocument->map(), exportDetails.mFileName);
+    if (!exportResult)
+    {
+        QMessageBox::critical(this, tr("Error Exporting Map!"),
+                              exportDetails.mFormat->errorString());
+    } else {
+        // Remember export parameters, so subsequent exports can be done faster
+        mapDocument->setLastExportFileName(exportDetails.mFileName);
+        mapDocument->setExportFormat(exportDetails.mFormat);
+    }
+}
+
+void MainWindow::exportTilesetAs(TilesetDocument *tilesetDocument)
+{
+    QString fileName = tilesetDocument->fileName();
+    QString selectedFilter =
+            mSettings.value(QLatin1String("lastUsedExportFilter")).toString();
+    auto exportDetails = chooseExportDetails<TilesetFormat>(fileName,
+                                                    tilesetDocument->lastExportFileName(),
+                                                    selectedFilter,
+                                                    this);
+    if (!exportDetails.valid()) {
+        return;
+    }
```

Comment:
Coding style: for single-line bodies, leave out the brackets.

Reply:
Done